### ISSUE TITLE
Having difficulty copying build artefact from image into repository

### ISSUE THREAD

**Issue by eggbean** [pguyot/arm-runner-action#44]
I cannot figure out how to copy a built compiled binary from the image to the cloned repository. I am building statically linked tmux binaries for armv7l and aarch64. This is the relevant part of the action yaml:

```
jobs:
  build_and_upgrade:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        arch: [armv7l, aarch64]
        include:
        - arch: armv7l
          cpu: cortex-a7
          base_image: raspios_lite:latest
        - arch: aarch64
          cpu: cortex-a53
          base_image: raspios_lite_arm64:latest

    steps:
    - uses: actions/checkout@v3
      with:
        token: ${{ secrets.github_token }}

    - name: Upgrade buildscript
      run: wget https://github.com/mjakob-gh/build-static-tmux/raw/master/build-static-tmux.sh -O bin/scripts/build-static-tmux.sh

    - name: Build on arm VM
      uses: pguyot/arm-runner-action@v2
      with:
        base_image: ${{ matrix.base_image }}
        cpu: ${{ matrix.cpu }}
        cpu_info: ${{ matrix.cpu_info }}
        copy_artifact_path: /tmp/tmux-static/bin/tmux
        copy_artifact_dest: bin/${{ matrix.arch }}
        image_additional_mb: 1024
        commands: |
            test $(uname -m) = ${{ matrix.arch }}
            /bin/sh bin/scripts/build-static-tmux.sh
            mv /tmp/tmux-static/bin/tmux bin/${{ matrix.arch }}
```

And here is the relevant part of the log:

```
2022-07-21T02:12:51.3566782Z tmux 3.3
2022-07-21T02:12:51.3567123Z ------------------
2022-07-21T02:12:51.8653604Z Downloading...[0;32m[OK][0m
2022-07-21T02:12:52.0018275Z Extracting....[0;32m[OK][0m
2022-07-21T02:14:31.1097454Z Configuring...[0;32m[OK][0m
2022-07-21T02:22:51.7513307Z Compiling.....[0;32m[OK][0m
2022-07-21T02:22:53.4481238Z Installing....[0;32m[OK][0m
2022-07-21T02:22:53.5799107Z Stripping.....[0;32m[OK][0m
2022-07-21T02:22:53.5800598Z 
2022-07-21T02:22:53.5800903Z Resulting files:
2022-07-21T02:22:53.5857343Z ----------------
2022-07-21T02:22:53.5860753Z Standard tmux binary:   /tmp/tmux-static/bin/tmux.linux-armv7l.gz
2022-07-21T02:22:53.5861215Z Stripped tmux binary:   /tmp/tmux-static/bin/tmux.linux-armv7l.stripped.gz
2022-07-21T02:22:54.2979121Z 
2022-07-21T02:22:54.2980214Z ----------------------------------------
2022-07-21T02:22:54.3311163Z Duration: 0h 59m 21s
2022-07-21T02:22:54.3311995Z 
2022-07-21T02:22:54.3321505Z + mv /tmp/tmux-static/bin/tmux bin/armv7l
2022-07-21T02:22:54.3694969Z + rc=0
2022-07-21T02:22:54.3695987Z + '[' -f /home/actions/temp/arm-runner/mnt/tmp/github_env.sh ']'
2022-07-21T02:22:54.3696606Z + exit 0
2022-07-21T02:22:54.3747430Z ##[group]Run case "yes" in
2022-07-21T02:22:54.3747710Z [36;1mcase "yes" in[0m
2022-07-21T02:22:54.3747906Z [36;1myes|true)[0m
2022-07-21T02:22:54.3748118Z [36;1m    set -x[0m
2022-07-21T02:22:54.3748315Z [36;1m;;[0m
2022-07-21T02:22:54.3748489Z [36;1mesac[0m
2022-07-21T02:22:54.3748703Z [36;1mrepository_path=[0m
2022-07-21T02:22:54.3748951Z [36;1mif [ "${repository_path}x" = "x" ]; then[0m
2022-07-21T02:22:54.3749259Z [36;1m    repository_name=`basename /home/runner/work/.dotfiles/.dotfiles`[0m
2022-07-21T02:22:54.3749578Z [36;1m    repository_path=/${repository_name}[0m
2022-07-21T02:22:54.3749889Z [36;1mfi[0m
2022-07-21T02:22:54.3750156Z [36;1mif [ "/tmp/tmux-static/bin/tmuxx" != "x" ] && [ "bin/armv7lx" != "x" ]; then[0m
2022-07-21T02:22:54.3750516Z [36;1m    while read -d\; copy_artifact_path; do[0m
2022-07-21T02:22:54.3750912Z [36;1m        [ -z "${copy_artifact_path}" ] || sudo cp -Rp /home/actions/temp/arm-runner/mnt${repository_path}/${copy_artifact_path} bin/armv7l[0m
2022-07-21T02:22:54.3751297Z [36;1m    done <<< "/tmp/tmux-static/bin/tmux;"[0m
2022-07-21T02:22:54.3751514Z [36;1mfi[0m
2022-07-21T02:22:54.3847394Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2022-07-21T02:22:54.3847659Z ##[endgroup]
2022-07-21T02:22:54.3951040Z + repository_path=
2022-07-21T02:22:54.3951851Z + '[' x = x ']'
2022-07-21T02:22:54.3954368Z ++ basename /home/runner/work/.dotfiles/.dotfiles
2022-07-21T02:22:54.3967979Z + repository_name=.dotfiles
2022-07-21T02:22:54.3968528Z + repository_path=/.dotfiles
2022-07-21T02:22:54.3969119Z + '[' /tmp/tmux-static/bin/tmuxx '!=' x ']'
2022-07-21T02:22:54.3975654Z + '[' bin/armv7lx '!=' x ']'
2022-07-21T02:22:54.4025753Z + read '-d;' copy_artifact_path
2022-07-21T02:22:54.4026704Z + '[' -z /tmp/tmux-static/bin/tmux ']'
2022-07-21T02:22:54.4027908Z + sudo cp -Rp /home/actions/temp/arm-runner/mnt/.dotfiles//tmp/tmux-static/bin/tmux bin/armv7l
2022-07-21T02:22:54.4111627Z cp: cannot stat '/home/actions/temp/arm-runner/mnt/.dotfiles//tmp/tmux-static/bin/tmux': No such file or directory
2022-07-21T02:22:54.4128230Z ##[error]Process completed with exit code 1.
2022-07-21T02:22:54.4362897Z Post job cleanup.
```

The buildscript is supposed to make a tmux binary at `/tmp/tmux-static/tmux` which I want to copy to `bin/armv7l/` in the repository, but it cannot find it.

Thanks.

**Comment by pguyot** [comment]
`copy_artifact_path` is a relative path in the emulated environment. It cannot be absolute as shown by
your log. `copy_artifact_dest` can be absolute and is a path in the "host" environment.

You probably want to swap the values of both parameters.

**Comment by eggbean** [comment]
Thanks. I was confused about you saying that I probably want to swap the parameters. I don't know why you thought that, but after experimenting and finally installing `tree` and searching through the whole filesystem, I realised that my variables should be this:

    copy_artifact_path: ../tmp/tmux-static/bin/tmux
    copy_artifact_dest: bin/${{ matrix.arch }}

Everythings is working fine, but I realised that I cannot use a jobs matrix and have the artefacts all get copied to the repository, so I do the builds sequentially. The x86_64 build takes 3 minutes, but the arm build take 62 and 81 mins, so the whole workflow takes 2hrs and 40mins.

This isn't a big deal, but I just want to make sure that this is to be expected or if there might be something wrong that I've done?
